src/spaceObjects/playerSpaceship.cpp

```diff
@@ -1972,9 +2011,19 @@ void PlayerSpaceship::commandCustomFunction(string name)
     sendClientCommand(packet);
 }
 
-void PlayerSpaceship::commandSetScienceLink(int32_t id){
+void PlayerSpaceship::commandSetScienceLink(P<ScanProbe> probe)
+{
+    sf::Packet packet;
+    packet << CMD_SET_SCIENCE_LINK;
+    packet << probe->getMultiplayerId();
```

Comment:
What if probe is "nullptr"? `player:commadnSetScienceLink(nil)` would do that I think.